### ISSUE TITLE
Training on CIFAR100, I get 0.976 error

### ISSUE THREAD

**Issue by litgm** [gpleiss/temperature_scaling#2]
Hi

I cloned your pakage, and ran it on my machine (with small modifications to pytorch 0.3 - added "deterministic" flag on conv)

Using the default parameters (depth=40, growth_rate=12, n_epochs=300, batch_size=64, lr=0.1, momentum=0.9), the final result is: 
Eval: (Epoch 300 of 300) [0079/0079]  Time: 0.00806 (1.999)  Loss: 0.19664 (1.823)  **Error: 0.87500 (0.976)**

**Comment by gpleiss** [comment]
I'm not sure what the issue is, but this package is only designed how perform temperature scaling.

**Comment by DerekArrow** [comment]
@litgm I think there is something wrong with the method to calculate 'error'. The predicted label calculated by topk() is in size 128x1, but the size of label is 128. So, you should make them the same size then do torch.eq()

**Comment by gpleiss** [comment]
@DerekArrow that sounds like it's very possible! The demo script hasn't been updated since the latest PyTorch update. Thanks for the catch!
